palace.pyx

```diff
@@ -302,7 +302,20 @@ cdef class Device:
         """
         self.impl.resume_dsp()
 
-    # TODO: get clock time
+    @property
+    def clock_time(self) -> int:
+        """Current clock time for the device.
+        This starts relative to the device being opened,
+        and does not increment while there are no
+        contexts nor while processing is paused.
+        This is currently based on std::chrono::steady_clock,
+        and so may not exactly match the rate that
+        sources play at.
+        In the future it may utilize an OpenAL extension to retrieve
+        the audio device's real clock which may tic at a subtly
+        different rate than the main clock(s).
+        """
```

Comment:
See [PEP 257](https://www.python.org/dev/peps/pep-0257/), the recommended length is 72 and paragraphs are separated by a line.
Also please rephrase for Python context.

Reply:
Since I just copied from alure's doc comments, I'm not sure if I should delete the whole notes that are related to C++ (...This is currently based on std::chrono::steady_clock...In the future it may utilize an OpenAL). It seems quite C++-specific to me.

Reply:
I think the docstring may be phrased as 

```
Current clock time for the device.

Notes
-----
This starts relative to the device being opened, and does not increment
while there are no contexts nor while processing is paused.  Currently,
this may not exactly match the rate that sources play at.  In the future
it may utilize an OpenAL extension to retrieve the audio device's real clock.
```